test/e2e/storage/testsuites/provisioning.go

```diff
@@ -247,7 +249,38 @@ func (p *provisioningTestSuite) DefineTests(driver storageframework.TestDriver,
 		manifests := []string{
 			"test/e2e/testing-manifests/storage-csi/any-volume-datasource/hello-populator-deploy.yaml",
 		}
-		popCleanup, err := storageutils.CreateFromManifests(f, f.Namespace, nil, manifests...)
+		popCleanup, err := storageutils.CreateFromManifests(f, f.Namespace,
+			func(item interface{}) error {
+				switch item := item.(type) {
+				case *appsv1.Deployment:
+					for i, container := range item.Spec.Template.Spec.Containers {
+						// framework.Logf("container name: %s", container.Name)
+						switch container.Name {
+						case "hello":
+							var found bool
+							args := []string{}
+							for _, arg := range container.Args {
+								if strings.HasPrefix(arg, "--namespace=") {
+									args = append(args, fmt.Sprintf("--namespace=%s", f.Namespace.Name))
```

Comment:
@bswartz @landreasyan I've create a PR for hello-populator to accept `--namespace` argument as  https://github.com/kubernetes-csi/lib-volume-populator/pull/17 and updated this PR to patch the hello-populator-deploy.yaml to add `--namespace` argument. 

If populator should be deployed to different namespace to the test namespace, we can change the code to create another namespace for populator and deploy it to the namespace.

Reply:
> If populator should be deployed to different namespace to the test namespace, we can change the code to create another namespace for populator and deploy it to the namespace.

The last commit "Create populator in a separate namespace" handles it
(and the current codes can be manually tested by the steps in https://github.com/kubernetes/kubernetes/pull/108664#issuecomment-1067365541).